### PR TITLE
Add TO Go post deliveryservice_user

### LINKED ISSUE #3836: Rewrite /deliveryservice_user to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/deliveryservice_user` is handled by Perl

## Expected / new behavior:
`/deliveryservice_user` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `POST`

## Minimal reproduction of the problem with instructions:
N/A

## Anything Else
PR in progress: #2457 . Progress to completion is unknown

### PR DESCRIPTION
**Pull request by rob05c**
Rewrites `/deliveryservice_user` to Go. Closes #3836

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1895/
Test PASSed.

**Comment by rob05c** [comment]
API tests are in https://github.com/apache/trafficcontrol/pull/2339

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1947/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1997/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2056/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2116/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4151/
Test PASSed.

File: lib/go-tc/deliveryservices.go [hunk 1]
```diff
@@ -557,6 +557,23 @@ type DeliveryServiceUserPost struct {
...
 
+func (dsu *DeliveryServiceUserPost) Validate(tx *sql.Tx) error {
+	errs := []string{}
+	if dsu.UserID == nil {
+		errs = append(errs, "userId missing")
+	}
+	if dsu.DeliveryServices == nil {
+		errs = append(errs, "deliveryServices missing")
+	}
+	if dsu.Replace == nil {
```
Comment (mhoppa) [inline lib/go-tc/deliveryservices.go]: the current api docs say this is optional - is that not true anymore?

**Comment by mhoppa** [comment]
This was decided to be deprecated as Delivery service users have no effect in our current system due to the introduction of tenancy. We can close this in place of opening a deprecation PR.

**Comment by mhoppa** [comment]
@ocket8888 @rob05c ^

**Comment by ocket8888** [comment]
Well if he wants he can still add deprecation notices to this. If that gets done before a deprecation PR is merged then we can merge this. If not, we can close this. Starting to look like that may be the case.

**Comment by mhoppa** [comment]
https://github.com/apache/trafficcontrol/pull/4369 replaces this PR

**Comment by ocket8888** [comment]
What he said

**Pull request closed without merging**